.dotnet/src/Custom/Chat/ChatResponseFormat.cs

```diff
@@ -18,28 +20,49 @@ namespace OpenAI.Chat;
 /// the model.
 /// </para>
 /// </remarks>
-[CodeGenModel("CreateChatCompletionRequestResponseFormat")]
-public partial class ChatResponseFormat
+[CodeGenModel("ChatResponseFormat")]
+public abstract partial class ChatResponseFormat
 {
-    // CUSTOM: Made internal.
-
-    /// <summary> Must be one of `text` or `json_object`. </summary>
-    [CodeGenMember("Type")]
-    internal InternalCreateChatCompletionRequestResponseFormatType? Type { get; set; }
+    /// <summary> text. </summary>
+    public static ChatResponseFormat Text { get; } = new InternalChatResponseFormatText();
+    /// <summary> json_object. </summary>
+    public static ChatResponseFormat JsonObject { get; } = new InternalChatResponseFormatJsonObject();
 
-    // CUSTOM: Made internal.
-    /// <summary> Initializes a new instance of <see cref="ChatResponseFormat"/>. </summary>
-    internal ChatResponseFormat()
+    public static ChatResponseFormat FromJsonSchema(
+        string name,
+        string description = null,
+        BinaryData jsonSchema = null,
+        bool? useStrictSchema = null)
```

Comment:
Since we're trying to avoid methods with optional parameters, I wonder if we should have a class that represents the JSON schema and pass it as a parameter.

Reply:
I thought our snapped guidance was "no optional parameters in constructors, all optional parameters in factories?"

Reply:
Let's revisit this question later just to make sure, but not block on it.